### PR TITLE
[FEATURE] CLI-PROJECT: check-config and upgrade Commands for V2-to-v3 Migration

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-200/GREAT-303
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/cli/toolkit.py [hunk 1]
```diff
@@ -565,73 +522,234 @@ def upgrade_project(
     cli_message(string=SECTION_SEPARATOR)
 
     # use loop in case multiple upgrades need to take place
-    while ge_config_version < CURRENT_GE_CONFIG_VERSION:
+    while int(ge_config_version) < CURRENT_GE_CONFIG_VERSION:
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/toolkit.py:525]: great catch :)

File: great_expectations/cli/toolkit.py [hunk 2]
```diff
@@ -565,73 +522,234 @@ def upgrade_project(
...
             pass
-    sys.exit(0)
+
+    (increment_version, exception_occurred,) = upgrade_project_one_version_increment(
+        context_root_dir=directory,
+        ge_config_version=ge_config_version,
+        continuation_message=EXIT_UPGRADE_CONTINUATION_MESSAGE,
+        update_version=False,
+        from_cli_upgrade_command=from_cli_upgrade_command,
+    )
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/toolkit.py]: curious about this part : why does the `upgrade_project_zero_versions_increment()` function need to call `upgrade_project_one_version_increment()`?

File: great_expectations/cli/upgrade_helpers/__init__.py [hunk 3]
```diff
@@ -4,4 +4,5 @@
 GE_UPGRADE_HELPER_VERSION_MAP = {
     1: UpgradeHelperV11,
     2: UpgradeHelperV13,
+    3: UpgradeHelperV13,  # Ensures that Manual Steps are highlighted (even if configuration is already at version 3.0).
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/upgrade_helpers/__init__.py:7]: comment really helps here. Thanks

File: tests/test_fixtures/upgrade_helper/great_expectations_v20_project_with_v30_configuration_and_v20_checkpoints/checkpoints/titanic_checkpoint_1.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#
+# For more details please see
+# https://docs.greatexpectations.io/en/latest/how_to_guides/validation/how_to_add_validations_data_or_suites_to_a_checkpoint.html
+validation_operator_name: action_list_operator
+# Batches are a list of batch_kwargs paired with a list of one or more suite
+# names. A Checkpoint can have one or more batches. This makes deploying
+# Great Expectations in your pipelines easy!
+batches:
+  - batch_kwargs:
+      path: /Users/alexsherstinsky/Downloads/Titanic.csv
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/upgrade_helper/great_expectations_v20_project_with_v30_configuration_and_v20_checkpoints/checkpoints/titanic_checkpoint_1.yml]: do we want to keep these paths?

**Review by Shinnnyshinshin — COMMENTED** [review]
Thank you for this @alexsherstinsky :) I also noticed there were a lot of `.gitkeep` files that were included as part of this PR. Do we want to include them?

update: .gitkeep is used to allow github to push empty folders

File: great_expectations/cli/toolkit.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/cli/toolkit.py]: renamed

File: tests/test_fixtures/upgrade_helper/great_expectations_v20_project_with_v30_configuration_and_v20_checkpoints/checkpoints/titanic_checkpoint_1.yml (hunk 4 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/upgrade_helper/great_expectations_v20_project_with_v30_configuration_and_v20_checkpoints/checkpoints/titanic_checkpoint_1.yml]: cleaned up -- thanks for the great catch!

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great :) thank you @alexsherstinsky

**Pull request merged**